nexus_system_orchestrator/src/context.hpp

```diff
@@ -29,47 +29,267 @@
 #include <rclcpp_action/rclcpp_action.hpp>
 #include <rclcpp_lifecycle/lifecycle_node.hpp>
 
+#include <mutex>
+#include <optional>
+
 namespace nexus::system_orchestrator {
 
 class Context
 {
-public: using WorkOrder = common::WorkOrder;
-public: using WorkOrderGoalHandle =
+public:
+  using WorkOrder = common::WorkOrder;
+  using WorkOrderGoalHandle =
     rclcpp_action::ServerGoalHandle<nexus::endpoints::WorkOrderAction::ActionType>;
-public: using WorkcellTask = nexus_orchestrator_msgs::msg::WorkcellTask;
-public: using TaskState = nexus_orchestrator_msgs::msg::TaskState;
+  using WorkcellTask = nexus_orchestrator_msgs::msg::WorkcellTask;
+  using TaskState = nexus_orchestrator_msgs::msg::TaskState;
+
+  Context(rclcpp_lifecycle::LifecycleNode& node) : _node(node) {}
+
+  rclcpp_lifecycle::LifecycleNode& get_node() const
+  {
+    std::lock_guard<std::mutex> lock(_mutex);
+    return _node;
+  }
+
+  Context& set_job_id(const std::string& job_id)
+  {
+    std::lock_guard<std::mutex> lock(_mutex);
+    _job_id = job_id;
+    return *this;
+  }
+
+  std::string get_job_id() const
+  {
+    std::lock_guard<std::mutex> lock(_mutex);
+    return _job_id;
+  }
+
+  Context& set_work_order(const WorkOrder& wo)
+  {
+    std::lock_guard<std::mutex> lock(_mutex);
+    _wo = wo;
+    return *this;
+  }
+
+  WorkOrder get_work_order() const
+  {
+    std::lock_guard<std::mutex> lock(_mutex);
+    return _wo;
+  }
+
+  Context& set_tasks(const std::vector<WorkcellTask>& tasks)
+  {
+    std::lock_guard<std::mutex> lock(_mutex);
+    _tasks = tasks;
+    return *this;
+  }
+
+  std::vector<WorkcellTask> get_tasks() const
+  {
+    std::lock_guard<std::mutex> lock(_mutex);
+    return _tasks;
+  }
+
+  std::size_t get_task_number() const
```

Comment:
Nit. `get_task_number` could mean something else.  How about `get_tasks_size` or `get_num_tasks` instead?
```suggestion
  std::size_t get_num_tasks() const
```

Reply:
Good call, done, https://github.com/osrf/nexus/pull/75/commits/1a073a5645b7fd2b0298ec130eb3845ed805dece